value/reflectcache.go

```diff
@@ -389,6 +391,11 @@ func unmarshal(data []byte, v interface{}) error {
 		if err := decoder.Decode(v); err != nil {
 			return err
 		}
+		next := decoder.InputOffset()
+		if _, err := decoder.Token(); !errors.Is(err, io.EOF) {
+			tail := bytes.TrimLeft(data[next:], " \t\r\n")
+			return fmt.Errorf("unexpected trailing data at offset %d", len(data)-len(tail))
+		}
```

Comment:
before adding more complexity here: any reason this function is not just like the following?
```golang
// unmarshal unmarshals the given data
// If v is a *map[string]interface{}, numbers are converted to int64 or float64
func unmarshal(data []byte, v interface{}) error {
	d := json.NewDecoder(bytes.NewReader(data))
	d.UseNumber()
	if err := d.Decode(&v); err != nil {
		return err
	}

	next := d.InputOffset()
	if _, err := d.Token(); !errors.Is(err, io.EOF) {
		tail := bytes.TrimLeft(data[next:], " \t\r\n")
		return fmt.Errorf("unexpected trailing data at offset %d", len(data)-len(tail))
	}

	// If the decode succeeds, post-process the map to convert json.Number objects to int64 or float64
	switch v := v.(type) {
	case *map[string]interface{}:
		return convertMapNumbers(*v, 0)
	case *[]interface{}:
		return convertSliceNumbers(*v, 0)
	default:
		return nil
	}
}
```

Reply:
It tests green.

Reply:
That seems like it would be equivalent in all the ways `unmarshal` is currently used. If a `v` with dynamic type `*interface{}` is passed, then JSON number inputs would unmarshal to `json.Number` with this change instead of `float64`.

Reply:
right, and I would claim this case was kind of an oversight. We should call convertNumber.

Reply:
What I mean is that it decoded to float64, while integers should become int64.